### PR TITLE
tests: sleep: convert legacy test to ztest

### PR DESCRIPTION
**Pull request by punitvara**
This patch do following things :
- fix checkpatch warnings
- replace conditions with ztest apis wherever necessary
- increase stack size to extra 256 to avoid failure of test case

Signed-off-by: Punit Vara <punit.vara@intel.com>

### CODE REVIEW

**Comment by punitvara** [comment]
@SavinayDharmappa Can you please review ?

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]
Why are you changing the stack size for the idle thread?? There is no reason why this has to be touched!

You should not have to change any thread stack sizes at all, increasing them like this is a regression!

**Comment by punitvara** [comment]
@andrewboie Thanks for the catch. I think I forgot to remove it after debugging the issue. Updated the commit

File: tests/kernel/sleep/src/sleep.c [hunk 1]
```diff
@@ -25,16 +26,16 @@
 #include <util_test_common.h>
 
 #if defined(CONFIG_ASSERT) && defined(CONFIG_DEBUG)
-#define THREAD_STACK    (384 + CONFIG_TEST_EXTRA_STACKSIZE)
+#define THREAD_STACK    (640 + CONFIG_TEST_EXTRA_STACKSIZE)
```
Comment (andrewboie) [inline tests/kernel/sleep/src/sleep.c]: why do we need to almost double the stack sizes?

**Comment by punitvara** [comment]
@andrewboie Can you please review this once ?

File: tests/kernel/sleep/src/sleep.c [hunk 2]
```diff
@@ -214,25 +220,27 @@ void main(void)
 	k_wakeup(test_thread_id);
 
 	if (test_failure) {
-		goto done_tests;
+		zassert_true(0, "test failure");
```
Comment (ramakrishnapallala) [inline tests/kernel/sleep/src/sleep.c]: please remove if block. you do the same wit zassert_* as well.

Comment (punitvara) [reply tests/kernel/sleep/src/sleep.c]: Done

**Review by ramakrishnapallala — APPROVED** [review]

**Comment by punitvara** [comment]
@nashif  Can you please merge this if it seems ok?

**Review by nashif — APPROVED** [review]

**Pull request merged**